diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/WhiteSpaceRule.kt

```diff
@@ -143,40 +164,74 @@ class WhiteSpaceRule : Rule("horizontal-whitespace") {
         }
     }
 
-    private fun handleBinaryOperator(node: ASTNode) {
-        if (node.elementType == COLON && node.treeParent.elementType != TYPE_CONSTRAINT) {
-            return
+    private fun handleColon(node: ASTNode) {
+        when (node.treeParent.elementType) {
+            CLASS, SECONDARY_CONSTRUCTOR, TYPE_CONSTRAINT, TYPE_PARAMETER, OBJECT_DECLARATION -> handleBinaryOperator(node)
+            VALUE_PARAMETER, PROPERTY -> handleToken(node, 0, 1)
+            // fixme: find examples or delete this line
+            else -> log.warn("Colon with treeParent.elementType=${node.treeParent.elementType}, not handled by WhiteSpaceRule")
         }
+    }
 
+    private fun handleBinaryOperator(node: ASTNode) {
         val operatorNode = if (node.elementType == OPERATION_REFERENCE) node.firstChildNode else node
 
         if (node.elementType == OPERATION_REFERENCE && node.treeParent.elementType.let { it == BINARY_EXPRESSION || it == POSTFIX_EXPRESSION } ||
                 node.elementType != OPERATION_REFERENCE) {
-            checkAndFixBinaryOperator(node, operatorNode)
+            val requiredNumSpaces = if (operatorNode.elementType in operatorsWithNoWhitespace) 0 else 1
+            handleToken(node, requiredNumSpaces, requiredNumSpaces)
         }
     }
 
-    private fun checkAndFixBinaryOperator(node: ASTNode, operatorNode: ASTNode) {
+    private fun handleToken(node: ASTNode, requiredSpacesBefore: Int?, requiredSpacesAfter: Int?) {
+        require(requiredSpacesBefore != null || requiredSpacesAfter != null)
         val spacesBefore = node.treePrev.numWhiteSpaces()
-        val spacesAfter = (node.treeNext ?: node.treeParent.treeNext)  // for `!!` and possibly other postfix expressions treeNext can be null
-                .numWhiteSpaces()
-        val requiredNumSpaces = if (operatorNode.elementType in operatorsWithNoWhitespace) 0 else 1
-        if (spacesBefore != null && spacesBefore != requiredNumSpaces || spacesAfter != null && spacesAfter != requiredNumSpaces) {
-            val freeText = "${node.text} should ${if (requiredNumSpaces == 0) "not " else ""}be surrounded by whitespaces"
+        val spacesAfter = if (requiredSpacesAfter != null) {
```

Comment:
Think more functional: 
`if ( != null) {} else null`
can be replaced by
```
requiredSpacesAfter?.map 
```

Reply:
Yes, will change.